### PR TITLE
Add guide link tracking on open navigation

### PR DESCRIPTION
**Pull request by mcgoooo**
This adds a click tracker on opening and closing guide navigation. 

This is needed to track usage of the open and close mechanism for template consolidating.

### CODE REVIEW

File: app/assets/javascripts/mobile.js [hunk 1]
```diff
@@ -30,11 +25,13 @@ $(function () {
         $this.text($this.text().replace(/Show/, 'Hide'));
         instance.$navbar.removeClass('page-navigation-closed');
         instance.$navbar.addClass('page-navigation-open');
+        GOVUK.analytics.trackEvent("contentsClicked", "openGuideNav",{});
```
Comment (nickcolley) [inline app/assets/javascripts/mobile.js]: This needs to be guarded like https://github.com/alphagov/static/blob/master/app/assets/javascripts/modules/track-click.js#L43 if the analytics module is not loaded

File: app/assets/javascripts/mobile.js [hunk 2]
```diff
@@ -14,11 +14,6 @@ $(function () {
 
   guideTabs.prototype = {
     init : function () {
-      // if the guideTabs are floated, quit (isn't small-screen)
```
Comment (nickcolley) [inline app/assets/javascripts/mobile.js]: Can we document why this is being removed?

Comment (nickcolley) [inline app/assets/javascripts/mobile.js]: Let's not change anything unless it impedes this tracking.

We're currently looking into the usefulness of this component within the work to merge Guides and Travel Advice into a Multipage template.

If we find users are fine with an inline list we can remove this entire component, if we decide we're going to keep this component then we'll clean this up and dedicate the time to that.

File: app/assets/javascripts/mobile.js (hunk 1 above)
Comment (nickcolley) [inline app/assets/javascripts/mobile.js]: Tiny nit: We should also have spaces between the parameters passed into `trackEvent`

**Comment by mcgoooo** [comment]
@fofr can you have a quick look at this please as @nickcolley is off

File: app/assets/javascripts/mobile.js [hunk 3]
```diff
@@ -18,23 +18,29 @@ $(function () {
...
-          instance = this;
+          instance = this,
+          hasAnalytics = GOVUK.analytics && GOVUK.analytics.trackEvent;
 
       if(!$this.hasClass('show-all-parts-open')) {
         $this.addClass('show-all-parts-open');
         $this.text($this.text().replace(/Show/, 'Hide'));
         instance.$navbar.removeClass('page-navigation-closed');
         instance.$navbar.addClass('page-navigation-open');
+        if hasAnalytics {
```
Comment (fofr) [inline app/assets/javascripts/mobile.js]: Should the if statement use parentheses for consistency?

Comment (mcgoooo) [inline app/assets/javascripts/mobile.js]: good spot, fixing now

File: app/assets/javascripts/mobile.js [hunk 4]
```diff
@@ -18,23 +18,29 @@ $(function () {
...
+          instance = this,
+          hasAnalytics = GOVUK.analytics && GOVUK.analytics.trackEvent;
 
       if(!$this.hasClass('show-all-parts-open')) {
         $this.addClass('show-all-parts-open');
         $this.text($this.text().replace(/Show/, 'Hide'));
         instance.$navbar.removeClass('page-navigation-closed');
         instance.$navbar.addClass('page-navigation-open');
+        if (hasAnalytics) {
+          GOVUK.analytics.trackEvent("contentsClicked", "openGuideNav", {});
```
Comment (fofr) [inline app/assets/javascripts/mobile.js]: The empty object can also be omitted.

File: app/assets/javascripts/mobile.js (hunk 3 above)
Comment (nickcolley) [inline app/assets/javascripts/mobile.js]: Consistency aside, It's not valid JavaScript: you end up with 'Uncaught SyntaxError: Unexpected identifier'.

**Comment by fofr** [comment]
Holding off merging until it's been tested.

**Comment by fofr** [comment]
@mcgoooo Did you get this tested ok? Can we merge?

**Pull request merged**